### DISCUSSION TITLE
Question about `<button-group>` subtree

### DISCUSSION THREAD

**Discussion by jonathantneal** [LeaVerou/nudeui#4] [Q&A]
Wow, this is a really wonderful project. The `<button-group>` component is an especially clear example of an advanced vanilla web component.

I’m curious about something, and I have no opinion to offer. In the `ButtonGroup` class, there’s a `MutationObserver` that I think is used to detect changes to inner `<button>` elements. While the examples make me expect `<button>` elements are strictly children, I also saw that the `subtree` is observed. Is this because descendent `<button>` elements are also supported? If so, have you run into instances where you would want to ignore a nested button?

https://github.com/LeaVerou/nudeforms/blob/main/button-group/button-group.js#L110

Again, I’m just curious. I haven’t found many projects as readable as this, so I guess I just want to be as influenced by this as possible. 😃

**Comment by LeaVerou** [comment]
Thanks for the kind words! 

I can't quite remember, but I think `subtree` was needed to observe attribute changes on children, otherwise the observer would only observe attribute changes on the button group itself. There's `childList`, but IIRC that just observes additions and removals of child nodes.
